src/storage/src/hummock/store/version.rs

```diff
@@ -67,17 +80,224 @@ pub struct HummockReadVersion {
 impl HummockReadVersion {
     /// Updates the read version with `VersionUpdate`.
     /// A `OrderIdx` that can uniquely identify the newly added entry will be returned.
-    pub fn update(&mut self, info: VersionUpdate) -> HummockResult<OrderIdx> {
-        unimplemented!()
+    pub fn update(&mut self, info: VersionUpdate) -> HummockResult<()> {
+        match info {
+            VersionUpdate::Staging(staging) => {
+                match staging {
+                    StagingData::ImmMem(imm) => self.staging.imm.push_back(imm),
```

Comment:
Seems that for new data we will do `push_back`. However, in the `get` below, we search the `imm` and `sst` in the forward order. Then we will first search the older imm, which may be incorrect.